subscriptions/fixed-price/checkout/node/src/server.ts

```diff
@@ -27,55 +33,114 @@ const endpointSecret = process.env.DIAGONAL_WEBHOOK_ENDPOINT_SECRET as string
+ *                   We recommend you pass a `reference` for what the user is paying for on checkout creation.
+ *                   Ideally this reference would refer to a unique product/plan/invoice id.
+ *
+ *
+ *  NOTE: `customer_id` and `reference` will be available in the `subscription.created` webhook event.
+ *         This will help you verify WHO subscribed and WHAT they paid for.
+ */
+app.post('/create-checkout-session/', async (req, res) => {
+  /* 
+     ***************************** Database code - rewrite yourself ********************************      
+      
+     Create a Diagonal `Customer` if not found in DB
 
-  res.sendStatus(200)
-})
+      ```
+        // Step 1: Get user from your DB
+        const user = UserTable.findOne({
+          req.user.id
+        })
 
-// Checkout sessions
-app.post('/create-checkout-session/', async (req, res) => {
-  // While creating a checkout session, you can pass in a customer ID
-  // to associate the session with a customer. This will allow you to
-  // retrieve the customer's subscriptions later.
-  let customerId = req.body.customerId
+        let diagonalCustomerId = user.diagonalCustomerId;
+
+        // Step 2: If user does not have a Diagonal customer id, create a new id and update user DB
+        if (diagonalCustomerId === undefined) {
+          
+          // Uniquely identify your customers
+          const diagonalCustomer = await diagonal.customers.create({
+            email: user.email,
+            name: user.name,
+            reference: user.id // optionally pass some unique ref if (email, name) is not unique
+          })
+
+          const user = UserTable.update(user.id, {
+            diagonalCustomerId: diagonalCustomer.id,
+          })
+
+          diagonalCustomerId = diagonalCustomer.id
+        }
+      ```
+  */
+
+  let customerId // diagonalCustomer
+
+  // If you are selling multiple products/service, pass the product id from your frontend
+  const plan = req.body.plan === 'basic' ? basicPlan : premiumPlan
 
   const checkoutSession = await diagonal.checkout.sessions.create({
     cancel_url: 'https://example.com/cancel',
     success_url: 'https://example.com/success',
-    amount: '10',
+    amount: plan.amount,
     payment_options: [
       {
-        tokens: ['usdc', 'dai'],
+        tokens: plan.tokens,
       },
     ],
     subscription: {
-      interval: 'month',
-      interval_count: 1,
+      interval: plan.interval,
+      interval_count: plan.intervalCount,
     },
-    customer_id: customerId,
+    customer_id: customerId, // Who is subscribing
+    reference: plan.id, // What they are paying for
   })
 
   res.redirect(checkoutSession.url)
 })
 
-// Subscriptions
 app.post('/upgrade-subscription/:id', async (req, res) => {
+  /*
+    You can upgrade a subscription by updating the subscription's amount and/or interval.
+    The subscription will be updated immediately, if a charge is required it will be automatically created.
```

Comment:
nit: The subscription will be updated immediately, and if a prorated charge is required, it will be created and processed automatically.